pkg/pkg/local.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2018 The ksonnet authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package pkg
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ksonnet/ksonnet/pkg/parts"
+
+	"github.com/ksonnet/ksonnet/pkg/app"
+	"github.com/ksonnet/ksonnet/pkg/prototype"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+// Local is a package based on vendored contents.
+type Local struct {
+	a              app.App
+	name           string
+	registryName   string
+	config         *parts.Spec
+	installChecker InstallChecker
+}
+
+var _ Package = (*Local)(nil)
```

Comment:
why do this? - looks like Google tells me that you get to force the compiler to make sure something is able to be cast as something else. but i don't understand when/why that is necessary that isn't covered by just using things normally or specifically implementing tests.

Reply:
It allows you to ensure a type implements a struct while editing it.